src/test/java/org/jenkinsci/plugins/workflow/EnvWorkflowTest.java

```diff
@@ -46,14 +46,15 @@
      */
     @Test public void isNodeNameAvailable() throws Exception {
         r.createSlave("node-test", "unix fast", null);
+        String builtInNodeLabel = r.jenkins.getSelfLabel().getExpression(); // compatibility with 2.307+
         WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "workflow-test");
 
         p.setDefinition(new CpsFlowDefinition(
-            "node('master') {\n" +
+            "node('" + builtInNodeLabel + "') {\n" +
             "  echo \"My name on master is ${env.NODE_NAME} using labels ${env.NODE_LABELS}\"\n" +
             "}\n",
             true));
-        r.assertLogContains("My name on master is master using labels master", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
+        r.assertLogContains("My name on master is master using labels " + builtInNodeLabel, r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
```

Comment:
This is awkward; I missed `NODE_NAME` being defined in https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/cffc780c39172277ecde560eeba25a833c419740/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java#L812 so Freestyle and Pipeline are out of sync now.

Reply:
https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/168